### PR TITLE
Add more methods to have_enqueued_job matcher

### PR DESCRIPTION
**Pull request by morgoth**
This is a follow up to https://github.com/rspec/rspec-rails/pull/1464

### CODE REVIEW

File: spec/rspec/rails/matchers/active_job_spec.rb [hunk 1]
```diff
@@ -144,5 +169,41 @@ def perform; end
...
+
+    it "passes with provided queue name" do
+      expect {
+        hello_job.set(:queue => "low").perform_later
+      }.to have_enqueued_job.on_queue("low")
+    end
+
+    it "passes with provided at date" do
+      expect {
+        hello_job.set(:wait_until => Date.tomorrow.noon).perform_later
```
Comment (soulcutter) [inline spec/rspec/rails/matchers/active_job_spec.rb]: Can you extract `Date.tomorrow.noon` into a variable? I know the window is small, but it's possible for this to fail when run precisely at midnight when the first call is before the date change and the second call is after.

File: spec/rspec/rails/matchers/active_job_spec.rb [hunk 2]
```diff
@@ -144,5 +169,41 @@ def perform; end
...
+    it "passes with provided arguments containing global id object" do
+      global_id_object = global_id_model.new(42)
+
+      expect {
+        hello_job.perform_later(global_id_object)
+      }.to have_enqueued_job.with(global_id_object)
+    end
+
+    it "generates failure message with all provided options" do
+      message = "expected to enqueue exactly 2 jobs, with [42], on queue low, at #{Date.tomorrow.noon}, but enqueued 0"
```
Comment (soulcutter) [inline spec/rspec/rails/matchers/active_job_spec.rb]: Extract the relative date into a variable here as well

**Comment by soulcutter** [comment]
This looks pretty good - would you be interested in adding cukes as well for documentation?

File: spec/rspec/rails/matchers/active_job_spec.rb (hunk 1 above)
Comment (morgoth) [inline spec/rspec/rails/matchers/active_job_spec.rb]: @soulcutter Done

**Comment by morgoth** [comment]
@soulcutter Yes, I was thinking about adding cukes already. I don't have experience with it so it may take some time. Should I try to add it in this PR or separate one is fine?

**Comment by soulcutter** [comment]
A separate PR is fine

**Comment by soulcutter** [comment]
Thanks for these enhancements! This is really coming together :+1:

**Pull request merged**